src/ParaTest/Runners/PHPUnit/ResultPrinter.php

```diff
@@ -201,16 +201,6 @@ public function printFeedback(ExecutableTest $test)
                 $test->getPath()
             ));
         }
-        if (!$reader->hasResults()) {
```

Comment:
Thanks a lot for your explanation. Of course I really want to get the groups thing working as well, so I'll try to think up some ways to "break" it in this branch and get back to you. 

On the other hand, instead of fixing this in the post-phase -- XML/log parsing -- I wonder if it could be fixed **up front** by simply excluding the entire test PHP file from even running if it is not in the requested group, or inversely has no tests NOT in the exclude group. 

Paratest actually does a PHP file parse before it even invokes any test, in order to get the test methods used in the file. The file is `\ParaTest\Parser\Parser`, which is used by the `initSuites` function in SuiteLoader.